toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -243,37 +243,32 @@ func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Netwo
 	const filePrefix = "10"
 
 	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
-		logger.Log.Errorf("Error accessing: %s", networkFileDir)
-		err = ferr
-		return
+		return fmt.Errorf("failed to access (%s):\n%w", networkFileDir, ferr)
 	} else if !exists {
-		err = fmt.Errorf("%s: no such path or directory", networkFileDir)
-		return
+		return fmt.Errorf("%s: no such path or directory", networkFileDir)
 	}
 
 	logger.Log.Debugf("Start creating network file")
 
 	networkFilePath := fmt.Sprintf("%s/%s-%s-%s.network", networkFileDir, filePrefix, networkData.BootProto, deviceName)
 	exists, err := file.PathExists(networkFilePath)
 	if err != nil {
-		logger.Log.Errorf("Error checking file path (%s): %s", networkFilePath, err)
-		return err
+		return fmt.Errorf("failed to check file path (%s):\n%w", networkFilePath, err)
 	} else if exists {
 		return fmt.Errorf("network file (%s) already exists", networkFilePath)
 	}
 
 	err = file.Create(networkFilePath, 0644)
 	if err != nil {
-		logger.Log.Errorf("Error creating file %s: %s", networkFilePath, err)
-		return
+		return fmt.Errorf("failed to create file (%s):\n%w", networkFilePath, err)
 	}
 
 	defer func() {
 		// Delete the network file on failure
 		if err != nil {
-			err = os.Remove(networkFilePath)
-			if err != nil {
-				logger.Log.Errorf("Failed to clean up network file (%s). Error: %s", networkFilePath, err)
+			cleanupErr := os.Remove(networkFilePath)
+			if cleanupErr != nil {
+				err = fmt.Errorf("failed to clean up network file (%s):\n%w\n%w", networkFilePath, cleanupErr, err)
```

Comment:
`err = fmt.Errorf("failed to clean up network file (%s):\n%w\ncleanup-error:\n%w", networkFilePath, err, cleanupErr)`

If you are going to add cleanup-errors to the error message, then you should clearly separate the original error from the cleanup error.

Reply:
how about
`err = fmt.Errorf("%w\nfailed to clean up network file (%s):\ncleanup-error:\n%w", err, networkFilePath, cleanupErr)`

Reply:
Oh, you're right. But possibly:

```golang
err = fmt.Errorf("%w\ncleanup-error: failed to clean up network file (%s):\n%w", err, networkFilePath, cleanupErr)
```